internal/provider/utils/utils.go

```diff
@@ -38,3 +34,24 @@ func GetHashedName(nsName string) string {
 	}
 	return fmt.Sprintf("%s-%s", resourceName, hashedName[0:8])
 }
+
+func HashString(str string) string {
+	h := sha256.New() // Using sha256 instead of sha1 due to Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
+	h.Write([]byte(str))
+	return strings.ToLower(fmt.Sprintf("%x", h.Sum(nil)))
+}
+
+// ExpectedContainerPortHashedName returns expected container port name with max length of 15 characters.
+// If mergedGateways is enabled or listener port name is larger than 15 characters it will return partially hashed name.
+// Listeners on merged gateways have a infraIR port name {GatewayNamespace}/{GatewayName}/{ListenerName}.
+func ExpectedContainerPortHashedName(name string) string {
+	if len(name) > 15 {
+		hashedName := HashString(name)
+		// replace `/` with `-` to create a valid K8s resource name
+		resourceName := strings.ReplaceAll(name, "/", "-")
+		listenerName := string(resourceName[2])
+
+		return fmt.Sprintf("%s-%s", listenerName, hashedName[0:14-len(listenerName)])
```

Comment:
this is going to be prone to collisions, id recommend hardcoding this to 7 chars of the listener and 8 chars of the hash.

Reply:
Yeah, probably it makes more sense, I don't like it either
Initially didn't want to make the listener name that short, I will change it like you proposed in the follow-up